### ISSUE TITLE
User-defined aggregates do not work with collections used for `initcond`

### ISSUE THREAD

**Issue by psarna** [scylladb/scylladb#10615]
In current code, `initcond` part of the aggregate is stored in system tables as `bytes` type, but the code erroneously serializes the initial values as strings, and then decodes from strings too. This is problematic, because CQL collections do not implement deserialization from strings. Instead, regular CQL serialization/deserialization routines should be used, because the column type is already ready for it.
Fortunately, UDAs are under an experimental flag, so the change can be carried out without new cluster features.

Current UDA code does not work, because `initcond` is serialized and deserialized as a string (due to its type definition in system schema tables), and we do not yet implement parsing collections from strings.

Example of a problematic line:
https://github.com/scylladb/scylla/blob/f873806c7c985a5716b2d6b8c7459d45985326a9/db/schema_tables.cc#L1638

Reported by @wmitros

**Comment by psarna** [comment]
Ugh, wait, I looked at some wrong code. `initcond` is unfortunately stored as `utf8` type, so a string:
```cql
cassandra@cqlsh> describe system_schema.aggregates;

CREATE TABLE system_schema.aggregates (
    keyspace_name text,
    aggregate_name text,
    argument_types frozen<list<text>>,
    final_func text,
    initcond text,
    return_type text,
    state_func text,
    state_type text,
    PRIMARY KEY (keyspace_name, aggregate_name, argument_types)
) WITH CLUSTERING ORDER BY (aggregate_name ASC, argument_types ASC)
    AND bloom_filter_fp_chance = 0.01
    AND caching = {'keys': 'ALL', 'rows_per_partition': 'ALL'}
    AND comment = 'user defined aggregate definitions'
    AND compaction = {'class': 'SizeTieredCompactionStrategy'}
    AND compression = {'sstable_compression': 'org.apache.cassandra.io.compress.LZ4Compressor'}
    AND crc_check_chance = 1.0
    AND dclocal_read_repair_chance = 0.0
    AND default_time_to_live = 0
    AND gc_grace_seconds = 604800
    AND max_index_interval = 2048
    AND memtable_flush_period_in_ms = 0
    AND min_index_interval = 128
    AND read_repair_chance = 0.0
    AND speculative_retry = '99.0PERCENTILE';
```
So perhaps we just need to implement `from_string` for tuples and other collections in order to remain compatible with C* :(

**Comment by wmitros** [comment]
This was fixed by #11250 but I forgot to mention it in the commit message

**Comment by avikivity** [comment]
Esoteric, not backporting.
